cpu/nrf52/periph/i2c.c

```diff
@@ -32,7 +32,13 @@
  */
 #define INVALID_SPEED_MASK  (0x0f)
 
-static mutex_t locks[I2C_NUMOF];
+/**
+ * @brief   Initialized bus locks (we have a maximum of two devices...)
+ */
+static mutex_t locks[] = {
+    MUTEX_INIT,
+    MUTEX_INIT
```

Comment:
Why is there two elements in the array ? What if we use only one I2C ? there is no need for the second lock then.

Reply:
that is correct. But the way the I2C devices are currently defined, there is no easy way to do on-demand static mutex initialization, so I rather waste 4 byte of memory here than to deal with potentially ugly code overhead. Keep in mind that something like `#if I2C_NUMOF > 1` does not work here...

Reply:
>  there is no easy way to do on-demand static mutex initialization

I fully agree with you as I encounter the same issue with #7588 
But I don't really like to waste memory with unused stuff.
Personally, I think we should init dynamically (but we must solve #8446 first)

Reply:
I agree with not wasting memory (in general)!

In this particular case though, I think we can live with it: this fix is rather temporary, as the driver is going to be slightly reworked once we finally start to tackle the adaption of #6576. With the changes coming there, we can easily go back to dynamic initialization of the mutex(es), as the init_master function will only be called once during system startup...

Reply:
>  this fix is rather temporary

Good to know.
>   once we finally start to tackle the adaption of #6576

Can't wait we start this one !

Reply:
>Can't wait we start this one !

@dylad happy to hear that :-)! You seem quite experienced so it would be great to have you on board for that. Bit off-topic: are you aware of any kind of reference I2C-slave device/implementation?